classes/Image.php

```diff
@@ -638,7 +638,12 @@ public function deleteImage($forceDelete = false)
         // Delete auto-generated images
         $image_types = ImageType::getImagesTypes();
         $sfContainer = SymfonyContainer::getInstance();
-        $isMultipleImageFormatFeatureEnabled = $sfContainer->get(FeatureFlagStateCheckerInterface::class)->isEnabled(FeatureFlagSettings::FEATURE_FLAG_MULTIPLE_IMAGE_FORMAT);
+
+        if ($sfContainer instanceof ContainerInterface) {
+	        $isMultipleImageFormatFeatureEnabled = $sfContainer->get('prestashop.core.admin.feature_flag.repository')->isEnabled(FeatureFlagSettings::FEATURE_FLAG_MULTIPLE_IMAGE_FORMAT);
+        }else {
+	        $isMultipleImageFormatFeatureEnabled = false;
```

Comment:
You must still use `FeatureFlagStateCheckerInterface` to retrieve the right state of the feature flag.
In fact, if you use the repository, you only retrieve the state in database, but now in `develop` branch, you can set feature flag states with env, dotenv or databse. So you must use the state checker to be sure to retrieve the right state.